nucliadb_node/src/replication/service.rs

```diff
@@ -117,6 +118,67 @@ async fn stream_data(
     Ok(())
 }
 
+async fn replicate_from_raw(
+    replica_state: RawReplicaState,
+    chunk_size: u64,
+    shard_path: &Path,
+    generation_id: &str,
+    _index_prefix: PathBuf,
```

Comment:
I find it a bit confusing that tantivy returns paths relative to the index and raw (vectors) returns paths relative to the shard. Would it be possible to change vectors so that it's also using paths relative to the index?

Feel free to say no, I saw something regarding `vectors` and `vectorset` which looks a bit tricky.

In any case, it would be nice to keep this in mind for `vectors2`

Reply:
You are totally right, the only reason why I did not unify the approach was to not overload the PR. I was trying to add the tantivy thing while changing the surroundings as little as possible. If you want I can unify in the same PR!